whereami/README.md

```diff
@@ -329,11 +331,99 @@ $ curl $ENDPOINT -s | jq .
+
+> Note: because gRPC is used as the protocol, the `whereami-grpc` reponse will omit any `header` fields *and* listens on port `9090` instead of port `8080`
+
+#### Step 1 - Deploy the whereami-grpc backend
+
+Deploy the `whereami-grpc` backend using the manifests from [k8s-grpc-backend-overlay-example](k8s-grpc-backend-overlay-example):
+
+```bash
+$ kubectl apply -k k8s-grpc-backend-overlay-example
+serviceaccount/whereami-grpc-ksa-backend created
+configmap/whereami-grpc-configmap-backend created
+service/whereami-grpc-backend created
+deployment.apps/whereami-grpc-backend created
+```
+
+#### Step 2 - Deploy the whereami-grpc frontend
+
+Now we're going to deploy the `whereami-grpc` frontend from the [k8s-grpc-frontend-overlay-example](k8s-grpc-frontend-overlay-example):
+
+```bash
+$ kubectl apply -k k8s-grpc-frontend-overlay-example
+serviceaccount/whereami-grpc-ksa-frontend created
+configmap/whereami-grpc-configmap-frontend created
+service/whereami-grpc-frontend created
+deployment.apps/whereami-grpc-frontend created
+```
+
+#### Step 3 - Query whereami-grpc frontend
+
+Get the external Service gRPC endpoint:
+
+```bash
+$ ENDPOINT=$(kubectl get svc whereami-grpc-frontend | grep -v EXTERNAL-IP | awk '{ print $4}')
+```
+
+Call the endpoint using [grpcurl](https://github.com/fullstorydev/grpcurl) to get the response. In this example, we use [jq](https://stedolan.github.io/jq/) to provide a little more structure to the response:
+
+```bash
+$ grpcurl -plaintext $ENDPOINT:9090 whereami.Whereami.GetPayload | jq .
+{
+  "backend_result": {
+    "clusterName": "asm-test-01",
+    "metadata": "grpc-backend",
+    "nodeName": "gke-asm-test-01-default-pool-02e1ecfb-1b9z.c.alexmattson-scratch.internal",
+    "podIp": "10.76.0.10",
+    "podName": "whereami-grpc-backend-f9b79888d-mcbxv",
+    "podNameEmoji": "🤒",
+    "podNamespace": "default",
+    "podServiceAccount": "whereami-grpc-ksa-backend",
+    "projectId": "alexmattson-scratch",
+    "timestamp": "2020-10-22T05:10:58",
+    "zone": "us-central1-c"
+  },
+  "cluster_name": "asm-test-01",
+  "metadata": "grpc-frontend",
+  "node_name": "gke-asm-test-01-default-pool-f9ad78c0-w3qr.c.alexmattson-scratch.internal",
+  "pod_ip": "10.76.1.10",
+  "pod_name": "whereami-grpc-frontend-88b54bc6-9w8cc",
+  "pod_name_emoji": "🇸🇦",
+  "pod_namespace": "default",
+  "pod_service_account": "whereami-grpc-ksa-frontend",
+  "project_id": "alexmattson-scratch",
+  "timestamp": "2020-10-22T05:10:58",
+  "zone": "us-central1-f"
+}
+```
 
 ### Notes
```

Comment:
This seems to be a recent upstream grpc bug from the release last week -- 
https://github.com/grpc/grpc/issues/24556 

I tried downgrading the grpc version but still got errors. I think using python-alpine may be the issue. https://github.com/grpc/grpc/issues/21483 

I switched the Dockerfile to use `python:3.8-slim` and grpc installed okay, but that broke the adduser alpine command so I removed that.   but then on startup, using my image, the pod says:

```
  Warning  Unhealthy  9s (x4 over 39s)   kubelet, gke-whereami-grpc-default-pool-6948884c-js4j  Readiness probe failed: OCI runtime exec failed: exec failed: container_linux.go:349: starting container process caused "exec: \"/bin/grpc_health_probe\": stat /bin/grpc_health_probe: no such file or directory": unknown
```

Reply:
For the health probe issue, that's because there's an additional step I documented for the buildpack approach but didn't back-port to the Dockerfile. Once I fix that I'll test out the `slim` vs `alpine` stuff.

Reply:
Dockerfile updated to pull down healthcheck binary *and* moved over to `slim`. Thanks for the suggestion @askmeegs